crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -142,19 +168,24 @@ impl<'a> ObjectRuntime<'a> {
     }
 
     pub fn new_id(&mut self, id: ObjectID) -> PartialVMResult<()> {
-        // Metered transactions don't have limits for now
-        if self.is_metered
-            && (self.state.new_ids.len() >= self.constants.max_num_new_move_object_ids)
-        {
-            return Err(PartialVMError::new(StatusCode::MEMORY_LIMIT_EXCEEDED)
-                .with_message(format!(
-                    "Creating more than {} IDs is not allowed",
-                    self.constants.max_num_new_move_object_ids
-                ))
-                .with_sub_status(
-                    VMMemoryLimitExceededSubStatusCode::NEW_ID_COUNT_LIMIT_EXCEEDED as u64,
-                ));
-        }
+        // If metered, we use the metered limit (non system tx limit) as the hard limit
+        // This macro takes care of that
+        match check_limit_by_meter!(
```

Comment:
it almost feels like the entire match could be in a macro, but it's all good I guess...

Reply:
Yes it seems that way, but I wasn't sure how the alerting logic would look, so I kept things explicit. If everything ends up looking similar, then I'll factor out the match. But good call out.